### PR TITLE
Implement `Arbitrary` for RTR payload types.

### PR DESCRIPTION
**Pull request by partim**
This PR adds implementations of the _arbitrary_ crate’s `Arbitrary` trait to some of the RTR payload types. This is necessary to support fuzzing.

The dependency on _arbitrary_ and consequently the impls are optional.

### CODE REVIEW

File: src/rtr/pdu.rs [hunk 1]
```diff
@@ -1007,6 +1020,18 @@ impl ProviderAsns {
...
 }
 
+#[cfg(feature = "arbitrary")]
+impl<'a> arbitrary::Arbitrary<'a> for ProviderAsns {
+    fn arbitrary(
+        u: &mut arbitrary::Unstructured<'a>
+    ) -> arbitrary::Result<Self> {
+        let size = usize::arbitrary(u)? % (
+            usize::from(u16::MAX) * mem::size_of::<Asn>()
+        );
```
Comment (DRiKE) [inline src/rtr/pdu.rs:1030]: This does not guarantee that `size` is a multiple of 4 (i.e. `mem::size_of::<Asn>()`), won't this result in invalid `ProviderAsns`?

File: src/uri.rs [hunk 2]
```diff
@@ -915,6 +915,96 @@ impl fmt::Display for Error {
...
+            Ok(Self {
+                bytes: res.into(),
+                module_start, path_start
+            })
+        }
+    }
+
+    impl<'a> Arbitrary<'a> for super::Https {
+        fn arbitrary(u: &mut Unstructured) -> arbitrary::Result<Self> {
+            let mut res = String::from("rsync://");
```
Comment (DRiKE) [inline src/uri.rs:950]:
```suggestion
            let mut res = String::from("https://");
```

File: src/uri.rs [hunk 3]
```diff
@@ -915,6 +915,96 @@ impl fmt::Display for Error {
...
+                path_idx
+            })
+        }
+    }
+
+    fn append_host(
+        res: &mut String, u: &mut Unstructured
+    ) -> arbitrary::Result<()> {
+        // Up to 255 characters of [.0-9A-Za-z-].
+        for _ in 0..u8::arbitrary(u)? {
```
Comment (DRiKE) [inline src/uri.rs:970]: host/authority can't be empty, right?
```suggestion
        for _ in 1..u8::arbitrary(u)? {
```

**Review by DRiKE — CHANGES_REQUESTED** [review]

File: src/rtr/pdu.rs (hunk 1 above)
Comment (partim) [reply src/rtr/pdu.rs:1030]: Just move the parentheses, right?

**Review by DRiKE — APPROVED** [review]

**Pull request merged**